Description:  Detect "armel" and "armhf" architectures on Debian
Last-Updated: 2017-05-03
Forwarded: yes
Bug-Origin: https://github.com/open-mpi/ompi/issues/2514

Index: openmpi-2.1.0rc2/config/opal_config_asm.m4
===================================================================
--- openmpi-2.1.0rc2.orig/config/opal_config_asm.m4
+++ openmpi-2.1.0rc2/config/opal_config_asm.m4
@@ -1027,7 +1027,7 @@ AC_DEFUN([OPAL_CONFIG_ASM],[
             OPAL_GCC_INLINE_ASSIGN='"mov %0, #0" : "=&r"(ret)'
             ;;
 
-        armv7*)
+        armv7*|arm-*-linux-gnueabihf)
             opal_cv_asm_arch="ARM"
             OPAL_ASM_SUPPORT_64BIT=1
             OPAL_ASM_ARM_VERSION=7
@@ -1046,7 +1046,7 @@ AC_DEFUN([OPAL_CONFIG_ASM],[
             OPAL_GCC_INLINE_ASSIGN='"mov %0, #0" : "=&r"(ret)'
             ;;
 
-        armv5*linux*|armv4*linux*)
+        armv5*linux*|armv4*linux*|arm-*-linux-gnueabi)
             # uses Linux kernel helpers for some atomic operations
             opal_cv_asm_arch="ARM"
             OPAL_ASM_SUPPORT_64BIT=0
